rust_snuba/src/processors/eap_spans.rs

```diff
@@ -0,0 +1,537 @@
+use anyhow::Context;
+use chrono::DateTime;
+use serde::Serialize;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+
+use crate::config::ProcessorConfig;
+use crate::processors::spans::FromSpanMessage;
+use crate::processors::utils::enforce_retention;
+use crate::types::{InsertBatch, KafkaMessageMetadata};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    _metadata: KafkaMessageMetadata,
+    config: &ProcessorConfig,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+    let msg: FromSpanMessage = serde_json::from_slice(payload_bytes)?;
+
+    let origin_timestamp = DateTime::from_timestamp(msg.received as i64, 0);
+    let mut span: EAPSpan = msg.try_into()?;
+
+    span.retention_days = Some(enforce_retention(span.retention_days, &config.env_config));
+
+    InsertBatch::from_rows([span], origin_timestamp)
+}
+
+#[derive(Debug, Default, Serialize)]
+struct EAPSpan {
+    // the span object for the new "events analytics platform"
+    organization_id: u64,
+    project_id: u64,
+    trace_id: Uuid,
+    span_id: u64,
+    #[serde(default)]
+    parent_span_id: u64,
+    segment_id: u64,      //aka transaction ID
+    segment_name: String, //aka transaction name
+    is_segment: bool,     //aka "is transaction"
```

Comment:
Those fields could be null in the future (span streaming) so let's add a `default` too.